### PR TITLE
Added demo update via BLE (GATT server on nRF52/RiotOS)

### PR DESCRIPTION
**Pull request by danielinux**

### CODE REVIEW

File: riotOS-nrf52840dk-ble/wolfboot.mk [hunk 1]
```diff
@@ -0,0 +1,71 @@
+ifdef WOLFBOOT_OFFSET
```
Comment (tomoveu) [inline riotOS-nrf52840dk-ble/wolfboot.mk:1]: is the WOLFBOOT_OFFSET setting relevant?

File: riotOS-nrf52840dk-ble/wolfboot.mk [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		$(COLOR_RESET)'
+	@$(COLOR_ECHO)
+
+
+$(WOLFBOOT_BIN):
+	@$(COLOR_ECHO) 'sources:'
+	@$(COLOR_ECHO) $(SRC)
+	@$(COLOR_ECHO)
+	make -C $(WOLFBOOT) clean
+	make -C $(WOLFBOOT) TARGET=nrf52 DEBUG=0 BOOT0_OFFSET=$(WOLFBOOT_OFFSET) \
```
Comment (tomoveu) [inline riotOS-nrf52840dk-ble/wolfboot.mk]: could this be a simpler like: make -C $(WOLFBOOT) BOOT0_OFFSET=$(WOLFBOOT_OFFSET) wolfboot.bin

File: riotOS-nrf52840dk-ble/wolfboot.mk [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		WOLFBOOT_PARTITION_BOOT_ADDRESS=0x20000 \
+		WOLFBOOT_PARTITION_UPDATE_ADDRESS=0x60000 \
+		WOLFBOOT_PARTITION_SWAP_ADDRESS=0xE0000 \
+		wolfboot.bin
+
+.PHONY: wolfboot-flash-bootloader wolfboot-flash
+
+wolfboot-flash-bootloader: HEXFILE = $(WOLFBOOT_BIN)
+wolfboot-flash-bootloader: $(WOLFBOOT_BIN) $(FLASHDEPS)
+	sudo $(FLASHER) $(FFLAGS) -o 0x0
```
Comment (tomoveu) [inline riotOS-nrf52840dk-ble/wolfboot.mk:55]: I could not find where FLASHER comes from, maybe add a comment from where it comes

**Review by tomoveu — COMMENTED** [review]
@danielinux good new example. I see most of it is Bluetooth code and the wolfBoot glue logic looks good. I got confused a bit with the Makefile(s), because the wolfboot.mk is invoked from within the app rather than the main Makefile. Other than that it looks good 👍

File: riotOS-nrf52840dk-ble/wolfboot.mk (hunk 2 above)
Comment (danielinux) [reply riotOS-nrf52840dk-ble/wolfboot.mk]: Correct. The configuration is passed through the config file, so those are redundant. Fixing.

File: riotOS-nrf52840dk-ble/wolfboot.mk (hunk 3 above)
Comment (danielinux) [reply riotOS-nrf52840dk-ble/wolfboot.mk:55]: `wolfboot.mk` is a module makefile for RIOT-OS. the `FLASHER` variable is set according to the target by RIOT-OS in `Makefile.include`

File: riotOS-nrf52840dk-ble/wolfboot.mk (hunk 1 above)
Comment (danielinux) [reply riotOS-nrf52840dk-ble/wolfboot.mk:1]: Yes, it's used by the wolfboot.mk to decide whether to change the offset of the linked application. When it's not defined, the application is linked to run from the entry point (no bootloader)

Comment (tomoveu) [reply riotOS-nrf52840dk-ble/wolfboot.mk:1]: Thanks for the explanation.

File: riotOS-nrf52840dk-ble/wolfboot.mk (hunk 3 above)
Comment (tomoveu) [reply riotOS-nrf52840dk-ble/wolfboot.mk:55]: Understood, it is RIOT-OS specific. Thanks

**Comment by tomoveu** [comment]
@danielinux thank you for the team work during the review, we caught one and the BLE update example is awesome!

**Pull request merged**